table/block_based_table_reader.cc

```diff
@@ -2110,6 +2112,7 @@ Status BlockBasedTable::Get(const ReadOptions& read_options, const Slice& key,
           if (!ParseInternalKey(biter.key(), &parsed_key)) {
             s = Status::Corruption(Slice());
           }
+          found = true;
```

Comment:
Here biter can seek to a key after `key`. Only when `SaveValue` return true we can be sure the key exists.

Reply:
SaveValue returns false if the key is not visible yet, or it needs the next value for merge, ... Independent of the return value of SaveValue, the fact that the key is present means that the bloom filter was correct about the presence of the key. The bloom filter's job is not to tell us if the key is usable as well.

Reply:
Yeah, but `biter` can seek to a key after `key` when `key` does not exist. So at this line you are still not sure if the key exist. If you look at `SaveValue` the first thing it does is check if the keys are equal.

Reply:
thanks for updating! how about this?

Reply:
oh i missed this comment. fixed that by counting it only if the key matches the user key.